### ISSUE TITLE
Optional range information for Markdown elements

### ISSUE THREAD

**Issue by artempyanykh** [fsprojects/FSharp.Formatting#742]
I'm looking into using the Markdown parser for a project, but noticed that the range information is optional for markdown elements. Glancing through the code I can see [only 1 place where the range can become None](https://github.com/fsprojects/FSharp.Formatting/blob/main/src/FSharp.Formatting.Markdown/MarkdownParser.fs#L1045-L1048) but this looks accidental. Am I missing something? Or the range indeed should be present for all elements in the _real world_ scenarios.

**Comment by dsyme** [comment]
@artempyanykh Yes, makes sense - please do submit a PR to make the ranges non-optional

**Comment by artempyanykh** [comment]
@dsyme thanks! I looked into it and it seems that something's off with some span ranges (potentially around links). It seems that markdown ranges need a bit more love than I can give ATM, but anyhow I filed a separate issue specifically around correctness of ranges in case things change #744.
